code/game/objects/items/weapons/RCD.dm

```diff
@@ -378,7 +403,188 @@ RCD
+					if(do_after(user, decondelay, target = A))
+						if(!useResource(deconcost, user)) return 0
+						activate()
+						qdel(A)
+						return TRUE
+				return FALSE
+		if(LIGHT_MODE)
+			if(iswallturf(A))
+				var/turf/closed/wall/W = A
+				if(checkResource(floorcost, user))
+					to_chat(user, "<span class='notice'>You start building a wall light...</span>")
+					user.Beam(A,icon_state="nzcrentrs_power",time=15)
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					playsound(src.loc, 'sound/effects/light_flicker.ogg', 50, 0)
+					if(do_after(user, floordelay, target = A))
+						if(!istype(W))
+							return FALSE
+						var/list/candidates = list()
+						var/turf/open/winner = null
+						var/winning_dist = null
+						for(var/direction in cardinal)
+							var/turf/C = get_step(W, direction)
+							var/list/dupes = checkdupes(C)
+							if(start.CanAtmosPass(C) && !dupes.len)
+								candidates += C
+						if(!candidates.len)
+							to_chat(user, "<span class='warning'>Valid target not found...</span>")
+							playsound(src.loc, 'sound/misc/compiler-failure.ogg', 30, 1)
+							return FALSE
+						for(var/turf/open/O in candidates)
+							if(istype(O))
+								var/x0 = O.x
+								var/y0 = O.y
+								var/contender = cheap_hypotenuse(start.x, start.y, x0, y0)
+								if(!winner)
+									winner = O
+									winning_dist = contender
+								else
+									if(contender < winning_dist) // lower is better
+										winner = O
+										winning_dist = contender
+						activate()
+						if(!useResource(wallcost, user))
+							return FALSE
+						var/light = get_turf(winner)
+						var/align = get_dir(winner, A)
+						var/obj/machinery/light/L = new /obj/machinery/light(light)
+						L.dir = align
+						L.color = color_choice
+						L.light_color = L.color
+						return TRUE
+				return FALSE
+
+			if(isfloorturf(A))
+				var/turf/open/floor/F = A
+				if(checkResource(floorcost, user))
+					to_chat(user, "<span class='notice'>You start building a floor light...</span>")
+					user.Beam(A,icon_state="nzcrentrs_power",time=15)
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					playsound(src.loc, 'sound/effects/light_flicker.ogg', 50, 1)
+					if(do_after(user, floordelay, target = A))
+						if(!istype(F)) return 0
+						if(!useResource(floorcost, user)) return 0
+						activate()
+						var/destination = get_turf(A)
+						var/obj/machinery/light/floor/FL = new /obj/machinery/light/floor(destination)
+						FL.color = color_choice
+						FL.light_color = FL.color
+						return TRUE
+				return FALSE
+
+		if(GLOW_MODE)
+			if(useResource(launchcost, user))
+				activate()
+				to_chat(user, "<span class='notice'>You fire a glowstick!</span>")
+				var/obj/item/device/flashlight/glowstick/G  = new /obj/item/device/flashlight/glowstick(start)
+				G.color = color_choice
+				G.light_color = G.color
+				G.throw_at(A, 9, 3, user)
+				G.on = TRUE
```

Comment:
I mean did you want the glowstick to start glowing when shot or when it hits something ?